docs/guide/acquiring_data/helioviewer.rst

```diff
@@ -37,45 +37,18 @@ using the get_datasources method::
     >>> # print a list of datasources and their associated ids
     >>> for observatory, instruments in datasources.items():  # doctest: +REMOTE_DATA
     ...     for inst, detectors in instruments.items():  # doctest: +REMOTE_DATA
-    ...         for det, measurements in detectors.items():  # doctest: +REMOTE_DATA
-    ...             for meas, params in measurements.items():  # doctest: +REMOTE_DATA
-    ...                 print("%s %s: %d" % (observatory, params['nickname'], params['sourceId']))  # doctest: +REMOTE_DATA
-    SOHO EIT 171: 0
-    SOHO EIT 195: 1
-    SOHO EIT 284: 2
-    SOHO EIT 304: 3
-    SOHO LASCO C2: 4
-    SOHO LASCO C3: 5
-    SOHO MDI Mag: 6
-    SOHO MDI Int: 7
-    SDO AIA 94: 8
-    SDO AIA 131: 9
-    SDO AIA 171: 10
-    SDO AIA 193: 11
-    SDO AIA 211: 12
-    SDO AIA 304: 13
-    SDO AIA 335: 14
-    SDO AIA 1600: 15
-    SDO AIA 1700: 16
-    SDO AIA 4500: 17
-    SDO HMI Int: 18
-    SDO HMI Mag: 19
-    STEREO_A EUVI-A 171: 20
-    STEREO_A EUVI-A 195: 21
-    STEREO_A EUVI-A 284: 22
-    STEREO_A EUVI-A 304: 23
-    STEREO_A COR1-A: 28
-    STEREO_A COR2-A: 29
-    STEREO_B EUVI-B 171: 24
-    STEREO_B EUVI-B 195: 25
-    STEREO_B EUVI-B 284: 26
-    STEREO_B EUVI-B 304: 27
-    STEREO_B COR1-B: 30
-    STEREO_B COR2-B: 31
-    PROBA2 SWAP 174: 32
-    Yohkoh SXT AlMgMn: 33
-    Yohkoh SXT thin-Al: 34
-    Yohkoh SXT white-light: 35
+    ...         for measurements in detectors():  # doctest: +REMOTE_DATA
```

Comment:
It took me way too long to track this down. The problem is this line. `detectors()` should be `detectors.items()`.

Reply:
I would also return the example back to the original. That should still work.

Reply:
So keeping the code same as I have written and just changing detectors to detectors.items()
will make this thing work?

Reply:
Yeah. But I don't understand why you had to change it. The original should still work

Reply:
It is not working on my system

Reply:
Neither the changed one nor the original one

Reply:
I got your version to work with that change I mentioned. But the original should still work. What's the error on your system?

Reply:
Traceback (most recent call last):
  File "sunp.py", line 9, in <module>
    print("%s %s: %d" % (observatory, measurements['nickname'], measurements['sourceId'])) 
TypeError: tuple indices must be integers or slices, not str

Reply:
I've that file on my local system to run the code

Reply:
That would imply that measurements['nickname'] is not a dictionary but a tuple. Probably need to for loop to be changed somewhere.

Reply:
If it is working for you then why is it showing an error on my system

Reply:
I've not tested the original yet. But your version with that tweak did work. 

The outcome of that example is to print all the source IDs. We still want that example to do that. Rewriting it to only print some of the IDs isn't a fix.

Reply:
Ok I will see again how to change it so that it works for me as well

Reply:
Thanks, and I would really want to know where am I wrong